api/openapi-spec.yaml

```diff
@@ -0,0 +1,473 @@
+          }
+      responses:
+        200:
+          description: Successfully retrieved the events
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  pageNo:
+                    type: integer
+                    example: 1
+                  pageSize:
+                    type: integer
+                    example: 500
+                  totalNumberItems:
+                    type: integer
+                    example: 234
+                  items:
+                    type: array
+                    items:
+                      type: object
+                      example: All found eiffel events
+        401:
+          description: Unauthorized
+          content: {}
+        403:
+          description: Forbidden
+          content: {}
+        404:
+          description: The requested events are not found
+          content: {}
+        500:
+          description: Internal server issue
+          content: {}
+  /events/{id}:
+    get:
+      tags:
+      - event-resource
+      summary: To get single event information
+      operationId: getEventUsingGET
+      parameters:
+      - name: id
+        in: path
+        description: "Id of the event."
+        required: true
+        schema:
+          type: string
+      - name: shallow
+        in: query
+        description: "Determines if external ER's should be used to compile the results of query. Use `false` to use External ER's."
+        schema:
+          type: boolean
+          default: false
+      - name: readable
+        in: query
+        description: |
+          Determines if event time should be in milliseconds, or in Human readable time format. Ex:
+
+          `false` 1499076742982
+
+          `true`  2018-10-31T13:36:00.824Z.
+        schema:
+          type: boolean
+          default: false
+      responses:
+        200:
+          description: Successfully retrieved the Event
+          content:
+            text/html:
+              schema:
+                type: string
+                example: "Webpage showing raw events but with clickable event links"
+            application/json:
+              schema:
+                type: object
+                example: The Eiffel event requested
+        400:
+          description: The eventId is not provided
+          content: {}
+        401:
+          description: Unauthorized
+          content: {}
+        403:
+          description: Forbidden
+          content: {}
+        404:
+          description: The requested event is not found
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  errorMsg:
+                    type: string
+                    example: "The requested event is not found"
+        500:
+          description: Internal server issue
+          content: {}
+  /search/{id}:
```

Comment:
The search endpoint only supports JSON whereas `/events/{id}` supports HTML also. If confusing this could be removed and added back later if deemed needed.

Reply:
Hmm. Yes, probably better to start with a tight spec.

Reply:
Removed it in a9debeb

Reply:
Removed this one in a9debeb